### DISCUSSION TITLE
Could adding iteration's fp as a new system leads to data leakages?

### DISCUSSION THREAD

**Discussion by dmh1998dmh** [deepmodeling/dpgen#376] [Q&A]
When a iteration finished, labelled data will be trained as a seperated system in the next iteration. DPGEN makes only one `set` in `02.fp/data.000` folder. Deepmd-kit documents said that deepmd trains mutlisystems one by one and chooses test frames randomly from dataset if there is only one `set `(this case).
I am wondering whether there is a data leakage in the new system and models be overfitted to the new system.
A learning lcurve is attached.
![捕获](https://user-images.githubusercontent.com/50235763/115833519-bcf87e80-a446-11eb-98da-fa964309e942.PNG)
The learning lcurve is not smooth, and is it possible sudden changes result from the overfit to addition dataset?

**Comment by AnguseZhang** [comment]
Learning curve not smooth: It's quite common since in a training process, systems are randomly selected and there's no guarantees for the monotonous decrease. 
Data leakage : Not fully understand what do you mean.  New labels are trained equally with old systems, if you do not use the "init-model" mode. You can use `dp test` to check your model quality.

**Reply by dmh1998dmh** [reply]
Thanks for your reply, What i mean in data leakage is that one label is used both in training and testing. One paper (DOI:10.1021/acs.chemrev.0c01111) said it would cause problems.
